ethereum/contracts/state-transition/chain-deps/StateTransitionChainStorage.sol

```diff
@@ -143,4 +108,9 @@ struct AppStorage {
     address admin;
     /// @notice Address that the governor or admin proposed as one that will replace admin role
     address pendingAdmin;
+    /// new fields
+    uint256 chainId;
+    address bridgehub;
+    address stateTransition;
+    bool upToDate;
```

Comment:
An addition to the comment about that forcing the rollups to update will take `O(state_transition_chains)` invocations. Note, that not only invocations, but actually writes, we'll have to update both:

- `upToDate` 
- `freeze`

So forcing the upgrades basically punishes the governor

Reply:
It does, but these writes will have to be done anyway. It is more efficient this way I think.